guava/src/com/google/common/math/Quantiles.java

```diff
@@ -603,22 +638,26 @@ private static void movePivotToStartOfSlice(double[] array, int from, int to) {
    * allRequired[i]} for {@code i} in the range [{@code requiredFrom}, {@code requiredTo}]. These
    * indexes must be sorted in the array and must all be in the range [{@code from}, {@code to}].
    */
+  @SuppressWarnings({"lowerbound:argument.type.incompatible",/* (1): Lowest possible `requiredFrom` value is 0.
+           Although `requiredBelow` can be negative, if `requiredBelow` is not >= `requiredFrom`, `required` can't be negative */
```

Comment:
This does not explain why the code is correct at all.
You'd need to show that `required` is positive on that line.
I think it relies on properties of the content of the arrays.